roles/autotested/tasks/produce_results.yml

```diff
@@ -10,13 +10,23 @@
         autotest_execution is defined and
         autotest_execution|success
 
+- debug:
+    msg: "I am {{ peon_metadata.name }}"
+  when: peon_metadata is defined
+
+# Extract a name unique to this run, but common across runs of same job
 - name: Attempt to run alternate results2junit
-  shell: "{{ autotest_docker_path }}/results2junit --name {{ inventory_hostname }} results/default > {{ autotest_dest_dir }}/client/results/default/results.junit"
+  shell: >
+    {{ autotest_docker_path }}/results2junit
+        --name {{ peon_metadata.name }}
+        results/default
+        > {{ autotest_dest_dir }}/client/results/default/results.junit
```

Comment:
I've fixed my script so it writes directly to results.junit in the same directory that's given as input arg. I will do the same in the python rewrite. So you can probably remove line 23 entirely.